File: Sharp.Data/StrikeData.cs
using Dapper;
using Npgsql;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Sharp.Data
{
    public class StrikeData
    {
        public List<Strike> GetStrikes(ulong guildId, ulong userId)
        {
            using var connection = DataExtentions.GetConnection();
            connection.Open();
            return connection.Query<Strike>($"select * from get_strikes('{guildId}', '{userId}')").ToList();
        }
        public async Task<List<Strike>> GetStrikesAsync(ulong guildId, ulong userId)
        {
            using var connection = DataExtentions.GetConnection();
            connection.Open();
            return await Task.FromResult(connection.QueryAsync<Strike>($"select * from get_strikes('{guildId}', '{userId}')").Result.ToList());
        }

        public void AddStrike(ulong guildId, ulong userId, ulong modId, string reason, string date)
        {
            using var connection = DataExtentions.GetConnection();
            connection.Open();
            connection.Execute($"select add_strike('{guildId}', '{userId}', '{modId}', '{reason}', '{date}')");
        }
        public async Task AddStrikeAsync(ulong guildId, ulong userId, ulong modId, string reason, string date)
        {
            using var connection = DataExtentions.GetConnection();
            connection.Open();
            await connection.ExecuteAsync($"select add_strike('{guildId}', '{userId}', '{modId}', '{reason}', '{date}')");
        }

        public void RemoveStrike(int strikeId)
        {
            using var connection = DataExtentions.GetConnection();
            connection.Open();
            connection.ExecuteAsync($"select remove_strike({strikeId})");
        }
        public async Task RemoveStrikeAsync(int strikeId)
        {
            using var connection = DataExtentions.GetConnection();
            connection.Open();
            await connection.ExecuteAsync($"select remove_strike({strikeId})");
        }

        public void RemoveAllStrikesFromUser(ulong guildId, ulong userId)
        {
            using var connection = DataExtentions.GetConnection();
            connection.Open();
            connection.ExecuteAsync($"select remove_all_strikes('{guildId}', '{userId}')");
        }
        public async Task RemoveAllStrikesFromUserAsync(ulong guildId, ulong userId)
        {
            using var connection = DataExtentions.GetConnection();
            connection.Open();
            await connection.ExecuteAsync($"select remove_all_strikes('{guildId}', '{userId}')");
        }
    }
}

File: Sharp.Data/StrikeData.fs
namespace Sharp.Data

open Sharp.Domain
open Npgsql.FSharp
open System

module StrikeData =

    let private parseStrike (read : RowReader) = 
        { id = read.int "id"
          guildId = read.string "guild_id" |> UInt64.Parse |> GuildId
          userId = read.string "user_id" |> UInt64.Parse |> UserId
          modId = read.string "mod_id" |> UInt64.Parse |> ModId
          reason = read.string "reason"
          date = read.timestamptz "date_time" }

    let addStrikeAsync (GuildId guildId) (UserId userId) (ModId modId) reason dateTime =
        sprintf "SELECT add_strike('%i', '%i', '%i', '%s', '%A')"
            guildId userId modId reason dateTime
        |> Operations.executeNonQueryAsync
          
    let getStrikesAsync (GuildId guildId) (UserId userId) =
        sprintf "SELECT * FROM strike
        	     WHERE guild_id = '%i' 
        	     AND user_id = '%i'"
            guildId userId
        |> Operations.executeQueryAsync parseStrike
         
    let getAllStrikesAsync (GuildId guildId) =
        sprintf "SELECT * FROM strike
        	     WHERE guild_id = '%i'" 
            guildId
        |> Operations.executeQueryAsync parseStrike

    let removeStrikeAsync (GuildId guildId) (strikeId : int) =
        sprintf "SELECT remove_strike('%i', '%i')"
            guildId strikeId
        |> Operations.executeNonQueryAsync
            
    let removeAllStrikesFromUserAsync (GuildId guildId) (UserId userId) =
        sprintf "SELECT remove_all_strikes('%i', '%i')"
            guildId userId
        |> Operations.executeNonQueryAsync